src/transformers/models/propainter/modeling_propainter.py

```diff
@@ -0,0 +1,4630 @@
+            self.bias,
+            self.stride,
+            self.padding,
+            self.dilation,
+            mask,
+        )
+
+        return hidden_states
+
+
+class ProPainterRecurrentFlowCompleteNet(nn.Module):
+    def __init__(self, config: ProPainterConfig):
+        super().__init__()
+        self.config = config
+        self.downsample = nn.Sequential(
+            nn.Conv3d(
+                3,
+                config.num_channels // 4,
+                kernel_size=(1, 5, 5),
+                stride=config.strides,
+                padding=(0, 2, 2),
+                padding_mode="replicate",
+            ),
+            nn.LeakyReLU(0.2, inplace=True),
+        )
+
+        self.encoder1 = nn.Sequential(
+            ProPainterP3DBlock(
+                config.num_channels // 4,
+                config.num_channels // 4,
+                config.patch_size,
+                config.strides[0],
+                config.padding,
+            ),
+            nn.LeakyReLU(0.2, inplace=True),
+            ProPainterP3DBlock(
+                config.num_channels // 4,
+                config.in_channels[0],
+                config.patch_size,
+                config.strides[1],
+                config.padding,
+            ),
+            nn.LeakyReLU(0.2, inplace=True),
+        )  # 4x
+
+        self.encoder2 = nn.Sequential(
+            ProPainterP3DBlock(
+                config.in_channels[0],
+                config.in_channels[0],
+                config.patch_size,
+                config.strides[0],
+                config.padding,
+            ),
+            nn.LeakyReLU(0.2, inplace=True),
+            ProPainterP3DBlock(
+                config.in_channels[0],
+                self.config.num_channels,
+                config.patch_size,
+                config.strides[1],
+                config.padding,
+            ),
+            nn.LeakyReLU(0.2, inplace=True),
+        )  # 8x
+
+        self.intermediate_dilation = nn.Sequential(
+            nn.Conv3d(
+                self.config.num_channels,
+                self.config.num_channels,
+                config.kernel_size_3d,
+                config.stride_3d,
+                padding=(0, 3, 3),
+                dilation=(1, 3, 3),
+            ),  # p = d*(k-1)/2
+            nn.LeakyReLU(0.2, inplace=True),
+            nn.Conv3d(
+                self.config.num_channels,
+                self.config.num_channels,
+                config.kernel_size_3d,
+                config.stride_3d,
+                padding=(0, 2, 2),
+                dilation=(1, 2, 2),
+            ),
+            nn.LeakyReLU(0.2, inplace=True),
+            nn.Conv3d(
+                self.config.num_channels,
+                self.config.num_channels,
+                config.kernel_size_3d,
+                config.stride_3d,
+                padding=(0, 1, 1),
+                dilation=(1, 1, 1),
+            ),
+            nn.LeakyReLU(0.2, inplace=True),
+        )
```

Comment:
would be nice to have all of this configurable!

Reply:
done :+1: